Mathlib/Combinatorics/Digraph/Basic.lean

```diff
@@ -0,0 +1,252 @@
+/-
+Copyright (c) 2024 Kyle Miller, Jack Cheverton. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kyle Miller, Jack Cheverton
+-/
+import Mathlib.Order.CompleteBooleanAlgebra
+import Mathlib.Data.Fintype.Pi
+import Mathlib.Tactic.ApplyFun
+
+/-!
+# Digraphs
+
+This module defines directed graphs on a vertex type `V`.
+
+## Main definitions
+
+* `Digraph` is a structure for the relation
+
+* `CompleteAtomicBooleanAlgebra` instance: Under the subgraph relation, `Digraph` forms a
+  `CompleteAtomicBooleanAlgebra`. In other words, this is the complete lattice of spanning subgraphs
+  of the complete graph.
+
+## Todo
+
+* The implementation of digraphs is currently incomplete. It was originally created by Kyle Miller
+using an older version of Mathlib. This version of the module is being ported into the current
+version of Mathlib by Jack Cheverton. Furthermore, new additions to the module are being made
+based on what has been added to SimpleGraph since the original implementation of Digraph was
+created.
+-/
+
+open Finset Function
+
+universe u v w
+
+/-- A directed graph is a relation `Adj` on a vertex type `V`.
+The relation describes which pairs of vertices are adjacent.
+-/
+@[ext]
+structure Digraph (V : Type u) where
```

Comment:
Is this the same as Quiver?

Reply:
Yes, and it's not a class and not meant to be used as a class.

The main function of `Digraph` is to use graph theory terminology for a basic concept.

Reply:
Possibly we could have a comment somewhere to this effect, but it's not essential.